cmd/gapit/stats.go

```diff
@@ -112,6 +112,57 @@ func (verb *infoVerb) getEventsInRange(ctx context.Context, client service.Servi
 	return events[begin:end], nil
 }
 
+type aggregateStats struct {
+	average float64
+	stddev  float64
+	median  int
+}
+
+func computeStats(values []int) aggregateStats {
+	if len(values) == 0 {
+		return aggregateStats{}
+	}
+	total, total2 := 0, 0
+
+	sort.Ints(values)
+
+	for _, v := range values {
+		total += v
+		total2 += v * v
+	}
+
+	n := float64(len(values))
+	avg, avg2 := float64(total)/n, float64(total2)/n
+	return aggregateStats{
+		average: avg,
+		stddev:  math.Sqrt(avg2 - avg*avg),
+		median:  values[len(values)/2],
+	}
+
+}
+
+func callsPerFrame(events []*service.Event) aggregateStats {
+	frameIndices := []uint64{}
+	for _, e := range events {
+		if e.Kind == service.EventKind_FirstInFrame {
+			frameIndices = append(frameIndices, e.Command.Indices[0])
+		}
+	}
+
+	counts := make([]int, len(frameIndices))
+	frame := 0
+	for _, e := range events {
+		for frame+1 < len(frameIndices) && e.Command.Indices[0] >= frameIndices[frame+1] {
+			frame++
+		}
+		if e.Kind == service.EventKind_AllCommands {
+			counts[frame]++
+		}
+	}
```

Comment:
The reason I did it this way was because a EventKind_AllCommands event and a EventKind_FirstInFrame event are generated for the same command, e.g.
```
[454] EventKind_AllCommands
[455] EventKind_FirstInFrame
[455] EventKind_AllCommands
[456] EventKind_AllCommands
```
and I wanted to make sure it would count the AllCommands event in the right frame regardless of the ordering between the the AllCommands event and the FirstInFrame event, although looking at `resolve/events.go` I see the AllCommands event will always be after the first in frame event.  Is it reasonable to depend on that not changing?  If so then your fix sounds good, otherwise I could add LastInFrame to requested event types and use that to track things (since the LastInFrame event will always be before the first AllCommands in the next frame).

Reply:
Well observed - please feel free to add a comment to the `FirstInFrame` and `LastInFrame` events in `gapis/service/service.proto` that all other command events must come between these two.

Reply:
It doesn't actually seem like that currently holds in `gapis/resolve/events.go`, in fact it seems gapit SxS video specifically assumes FramebufferObservation is the very last.  I'll change it so that FirstInFrame will always be the first for now.

Reply:
Okay, fair enough. Feel free to keep as-is then. Maybe add a comment stating why this is done in multiple passes.